merlin.opam

```diff
@@ -51,8 +51,11 @@ Add opam emacs directory to your load-path by appending this to your .emacs:
     (add-hook 'tuareg-mode-hook 'merlin-mode t)
     (add-hook 'caml-mode-hook 'merlin-mode t)
     ;; Use opam switch to lookup ocamlmerlin binary
-    (setq merlin-command 'opam)))
-
+    (setq merlin-command 'opam)
+    ;; For a better UX you can instead write
+    ;; (setq merlin-command \"ocamlmerlin\")
```

Comment:
I don't see why `(setq merlin-command 'opam)` would be problematic when using `opam-switch-mode`.
- As far as I understand, `(setq merlin-command 'opam)` means that the path to the Merlin binary will be asked to opam, and opam, according to the currently selected switch, should answer whith the correct binary (regardless of the `$PATH`).
- Using `opam-switch-mode` allows one to easily change the active switch and load the correct `$PATH`. This means that Merlin can indeed be started by calling `ocamlmerlin`, but I don't see why it would prevent the opam-based detection to work ?

Reply:
@voodoos thanks for your comments!

> I don't see why (setq merlin-command 'opam) would be problematic when using opam-switch-mode.

> As far as I understand, (setq merlin-command 'opam) means that the path to the Merlin binary will be asked to opam, and opam, according to the currently selected switch, should answer whith the correct binary (regardless of the $PATH).

Yes and no: there is indeed a machinery in merlin@master to guess the Merlin binary when merlin-command = `'opam` **but** it is static: it relies on the current switch chosen at emacs startup, which implies the current behavior is buggy/distracting, see https://github.com/ocaml/merlin/issues/1591 as well as https://github.com/ocaml/merlin/issues/1591#issuecomment-1518894314 for the current workaround :-|

> Using opam-switch-mode allows one to easily change the active switch and load the correct $PATH. This means that Merlin can indeed be started by calling ocamlmerlin, but I don't see why it would prevent the opam-based detection to work ?

I didn't mean merlin's `'opam`-based static finding of ocamlmerlin wouldn't work, I meant opam-switch-mode's dynamic finding of ocamlmerlin by changing the `$PATH` **only** works if the minor mode lookups for a basename (roughly, emacs' equivalent of `$ which ocamlmerlin`)